testsuite/src/main/java/io/netty/testsuite/transport/socket/SocketHalfClosedTest.java

```diff
@@ -52,14 +52,147 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assumptions.assumeFalse;
 
+@Timeout(value = 20000, unit = MILLISECONDS)
 public class SocketHalfClosedTest extends AbstractSocketTest {
 
     protected int maxReadCompleteWithNoDataAfterInputShutdown() {
         return 2; // nio needs read flag to detect full closure.
     }
 
     @Test
-    @Timeout(value = 5000, unit = MILLISECONDS)
+    public void testAllDataReadEventTriggeredAfterHalfClosure(TestInfo testInfo) throws Throwable {
+        run(testInfo, new Runner<ServerBootstrap, Bootstrap>() {
+            @Override
+            public void run(ServerBootstrap serverBootstrap, Bootstrap bootstrap) throws Throwable {
+                // TODO: there needs to be a more robust way to detect this
+                if (bootstrap.toString().contains("NioSocketChannel") ||
+                        bootstrap.toString().contains("OioSocketChannel")) {
+                    return;
+                }
```

Comment:
This is admittedly very janky. There are a few options.
1. Enhance `BootstrapComboFactory`: this will require changing a number of tests.
2. Change the `Bootstrap` to be more inspectable. This expands surface API.
3. Move the test into its own abstract suite and extend in epoll, kqueue, and io_uring. This is easier to forget and this does feel like it belongs in this suite.
4. Keep it as it is. I wouldn't be proud of it but that's okay.

Reply:
You should be able to check if the group is an IoEventLoopGroup and if so cast to it and check if the NioIoHandler is used. For Oio you just check if it's of type OioEventLoopGroup

Reply:
That works, thank you @normanmaurer.